rustls/src/client/client_conn.rs

```diff
@@ -823,13 +850,14 @@ impl ConnectionCore<ClientConnectionData> {
         name: ServerName<'static>,
         extra_exts: Vec<ClientExtension>,
         proto: Protocol,
+        alpn_protocols: Vec<Vec<u8>>,
     ) -> Result<Self, Error> {
         let mut common_state = CommonState::new(Side::Client);
         common_state.set_max_fragment_size(config.max_fragment_size)?;
         common_state.protocol = proto;
         common_state.enable_secret_extraction = config.enable_secret_extraction;
         common_state.fips = config.fips();
-        let mut data = ClientConnectionData::new();
+        let mut data = ClientConnectionData::new(alpn_protocols);
```

Comment:
I suggest this data is instead passed into `hs::start_handshake`, so it can be deallocated as soon as possible rather than kept for the full life of the connection.

Reply:
@CodeMan62 Did I miss a detail, or is this feedback still unaddressed? I see `alpn_protocols` passed into the `ClientConnectionData::new()` call on L864 of `client_conn.rs`, in addition to the call to `hs::start_handshake()` on L873.